core-common/src/main/java/org/zalando/nakadi/util/CompressionBodyRequestFilter.java

```diff
@@ -44,18 +45,22 @@ public final void doFilter(final ServletRequest servletRequest, final ServletRes
 
         HttpServletRequest request = (HttpServletRequest) servletRequest;
 
-        final boolean isGzipped = Optional
-                .ofNullable(request.getHeader(CONTENT_ENCODING))
-                .map(encoding -> encoding.contains("gzip"))
-                .orElse(false);
-
-        if (isGzipped && !POST.matches(request.getMethod())) {
+        final Optional<String> contentEncodingOpt = Optional.ofNullable(
+                request.getHeader(CONTENT_ENCODING));
+        if (contentEncodingOpt.isPresent() && !HttpMethod.POST.matches(request.getMethod())) {
             reportNotAcceptableError((HttpServletResponse) servletResponse, request);
             return;
+        } else if (contentEncodingOpt.isPresent()) {
+            final String contentEncoding = contentEncodingOpt.get();
+            if (contentEncoding.contains("gzip")) {
+                request = new GzipServletRequestWrapper(request);
+            }
+
+            if (contentEncoding.contains("zstd")) {
+                request = new ZstdServletRequestWrapper(request);
+            }
```

Comment:
It will be better to use only one request wrapper (using `else if` construction)